sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -50,7 +50,10 @@ trait InvokeLike extends Expression with NonSQLExpression {
 
   def propagateNull: Boolean
 
-  protected lazy val needNullCheck: Boolean = propagateNull && arguments.exists(_.nullable)
+  protected lazy val needNullCheck: Boolean = needNullCheckForIndex.contains(true)
+  protected lazy val needNullCheckForIndex: Array[Boolean] =
+    arguments.map(a => a.nullable && (propagateNull ||
```

Comment:
For this change, I think the definition of `propagateNull` is somehow different now. Previously if `propagateNull` is true, null will be propagated, but now it also depends on if the argument is primitive type. It is better to update `propagateNull` param doc.

Reply:
Oh, nvm. I saw you updated it below.